Task BJS2-69161

MightySonnya opened the pull request:
1) Character class added;
2) Warrior class extends;
3) Archer class extends;

MightySonnya committed:
Task BJS2-69161

1)Character class added;
2)Warrior class extends;
3)Archer class extends;

20cognosce commented on src/main/java/school/faang/Archer.java:
```diff
@@ -0,0 +1,14 @@
+public class Archer extends Character {
+
+    public Archer(String name) {
+        super(name, 3, 10, 5);
```
магические числа
создай класс-конфигурацию и вынеси туда все public static final настройки для всех

20cognosce commented on src/main/java/school/faang/Archer.java:
```diff
@@ -0,0 +1,14 @@
+    }
+
+    public void attack(Character opponent) {
+        opponent.reduseHealth(this.healthPoint);
```
reduce

20cognosce commented on src/main/java/school/faang/Warrior.java:
```diff
@@ -0,0 +1,13 @@
+package school.faang;
+
+public class Warrior extends Character {
```
Аналогично

20cognosce commented on src/main/java/school/faang/Character.java:
```diff
@@ -0,0 +1,58 @@
+package school.faang;
+
+public abstract class Character {
+    protected String name;
+    protected int strength;
+    protected int agility;
+    protected int intellect;
```
вот эти все поля могут быть private final

20cognosce commented on src/main/java/school/faang/Character.java:
```diff
@@ -0,0 +1,58 @@
+
+    public Character(String name) {
+        this.name = name;
+        this.strength = 5;
+        this.agility = 5;
+        this.intellect = 5;
+        this.healthPoint = 100;
```
в конфигурацию тоже

20cognosce commented on src/main/java/school/faang/Character.java:
```diff
@@ -0,0 +1,58 @@
+        }
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public int getStrength() {
+        return strength;
+    }
+
+    public int getAgility() {
+        return agility;
+    }
+
+    public int getIntellect() {
+        return intellect;
+    }
+
+    public int getHealthPoint() {
+        return healthPoint;
+    }
```
Используй lobmok @Getter

20cognosce commented on src/main/java/school/faang/Character.java:
```diff
@@ -0,0 +1,58 @@
+        return healthPoint;
+    }
+}
+
+
+
```
Пустые строки не оставляем

20cognosce commented on src/main/java/school/faang/Character.java:
```diff
@@ -0,0 +1,58 @@
+    protected int strength;
+    protected int agility;
+    protected int intellect;
+    protected int healthPoint;
```
это можно private

20cognosce commented on src/main/java/school/faang/Character.java:
```diff
@@ -0,0 +1,58 @@
+    protected int intellect;
+    protected int healthPoint;
+
+    public Character(String name) {
```
конструкторы можно сделать protected

MightySonnya committed:
Task BJS2-69161

1) public static final added
2) delete empty lines
3) abstract class have private variables now

20cognosce commented on src/main/java/school/faang/Character.java:
```diff
@@ -0,0 +1,39 @@
+
+@Getter
+public abstract class Character {
+    private String name;
+    private int strength;
+    private int agility;
+    private int intellect;
```
final

20cognosce commented on src/main/java/school/faang/Character.java:
```diff
@@ -0,0 +1,39 @@
+
+    protected Character(String name) {
+        this.name = name;
+        this.strength = 5;
+        this.agility = 5;
+        this.intellect = 5;
```
почему тут остались магические числа?)

20cognosce reviewed (approved):
Пойдёт, давай дальше решать

20cognosce commented:
> Task :checkstyleTest NO-SOURCE
Error: eckstyle] [ERROR] /home/runner/work/god_bless/god_bless/src/main/java/school/faang/Warrior.java:16: Line is longer than 120 characters (found 121). [LineLength]

У тебя ошибка сборки из-за того, что код не прошел чекстайл

20cognosce commented:
![image](https://github.com/user-attachments/assets/b6a39b79-297b-41b7-85ba-72352e724429)
Поправь и можешь замерджить

MightySonnya committed:
Task BJS2-69161
1) final!!!

20cognosce commented:
Нет, Андрей, это еще не финал
Ты не поправил кодстайл

> Task :checkstyleMain
Error: eckstyle] [ERROR] /home/runner/work/god_bless/god_bless/src/main/java/school/faang/Warrior.java:17: Line is longer than 120 characters (found 121). [LineLength]

> Task :checkstyleMain FAILED

Видишь, что у тебя сборка красная? Зайди в неё и посмотри вывод в консоль гитхаба
![image](https://github.com/user-attachments/assets/3cfe074d-0f0e-401c-876e-4e2f4cff201b)

20cognosce commented:
Слишком много символов в одной строке

MightySonnya committed:
Task BJS2-69161
1) final!!!
